### ISSUE TITLE
[Duovero] Audio is broken

### ISSUE THREAD

**Issue by adam-lee** [gumstix/Gumstix-YoctoProject-Repo#8]
With the kernel 3.6.11, there is no audio output.

**Comment by adam-lee** [comment]
```
    Note on DuoVero audio input:
    
    DuoVero Parlor board has a TRRS audio socket. The intention was to allow both
    audio input and output with a single socket. Audio out works as intended.
    
    There are two complications when it comes to audio input unfortunately:
    
    1. DuoVero Parlor's audio jack is TRRS OMTP. Most modern earbuds and headphones
    including Apple iPhone and Google Android phones use CTIA. It means the ground
    and audio-in (Ring 2 and Sleeve) are reversed in the Parlor board when compared
    to modern devices.
    
    2. DuoVero Zepher and Crystal COMs use TI's TWL6040 as the audio codec. The
    headset-in is connected to audio-in pin of the Parlor's TRRS audio jack.
    However microphone bias (LDO on TWL6040) which should power a microphone is
    left unused, effectively rendering the audio-in as a line-in.
    
    In summary, the audio jack on Parlor board can only be used as audio-out. I do
    not believe there are microphones that are self-powered, and OMTP compatible.
    The length of the sleeve (ground) of all audio-in only jacks seem to be equal
    to the length of the sleeve (ground) and ring 2 (mic-in) combined in TRRS.
    
    When designing a custom board for DuoVero, make sure to add an external mic
    bias. Mic bias from TWL6040 is not available as it is a BGA package. And be
    sure to use CTIA.
```
